### PR TITLE
Add new meta

### PR DESCRIPTION
**Pull request by azoyan**

### CODE REVIEW

File: .metahash [hunk 1]
```diff
@@ -0,0 +1 @@
+[187, 253, 48, 20, 197, 233, 166, 18, 173, 174, 8, 49, 30, 95, 225, 204, 148, 25, 41, 168, 229, 200, 113, 40, 39, 234, 34, 146, 119, 212, 48, 155]
```
Comment (fluiderson) [inline .metahash:1]: Add this file to `.gitignore`. It's only needed during building.

File: Cargo.toml [hunk 2]
```diff
@@ -6,14 +6,26 @@ license = "MIT"
...
 [dependencies]
-gstd = { git = "https://github.com/gear-tech/gear.git", branch = "stable" }
+gstd = { git = "https://github.com/gear-tech/gear.git" }
 codec = { package = "parity-scale-codec", version = "3.2.1", default-features = false }
-ft-io = { git = "https://github.com/gear-dapps/fungible-token.git", tag = "0.1.3" }
+staking-io = { path = "io" }
+ft-io = { git = "https://github.com/gear-dapps/fungible-token.git", branch = "new_meta" }
 scale-info = { version = "2.3.1", default-features = false }
+hashbrown = "0.13.1"
+gmeta = { git = "https://github.com/gear-tech/gear", features = ["codegen"] }
```
Comment (fluiderson) [inline Cargo.toml:15]:
```suggestion
gmeta = { git = "https://github.com/gear-tech/gear" }
```
This feature only needed in metawasm crates.

File: Cargo.toml [hunk 3]
```diff
@@ -6,14 +6,26 @@ license = "MIT"
...
 
 [build-dependencies]
-gear-wasm-builder = { git = "https://github.com/gear-tech/gear.git", branch = "stable" }
+gear-wasm-builder = { git = "https://github.com/gear-tech/gear.git" }
+staking-io = { path = "io" }
+
+# [patch.crates-io]
+# # Two dependecies below goes for subxt v0.22.0
+# sp-core = { git = "https://github.com/gear-tech/substrate.git", branch = "gear-stable" }
+# sp-runtime = { git = "https://github.com/gear-tech/substrate.git", branch = "gear-stable" }
```
Comment (fluiderson) [inline Cargo.toml:31]:
```suggestion
```
Unused code?

File: Makefile [hunk 4]
```diff
@@ -12,28 +12,48 @@ clean:
...
 	@rustup target add wasm32-unknown-unknown --toolchain nightly
 
-linter:
-	@echo ──────────── Run linter ───────────────────────
+lint:
+	@echo ⚙️ Running the linter...
 	@cargo +nightly clippy --all-targets -- --no-deps -D warnings
 
-pre-commit: fmt linter test
+pre-commit: fmt lint test
```
Comment (fluiderson) [inline Makefile:31]:
```suggestion
pre-commit: fmt lint full-test
```

File: Makefile [hunk 5]
```diff
@@ -12,28 +12,48 @@ clean:
...
+	cat gear.pid;
+	@cargo test --package staking --test node_test -- --include-ignored --test-threads=1; 	kill `(cat gear.pid)`; rm gear; rm gear.pid
+
+full-test:
+	@if [ ! -f "./target/fungible_token-0.1.3.wasm" ]; then\
+		wget "https://github.com/gear-dapps/fungible-token/releases/download/0.1.3/fungible_token-0.1.3.wasm"\
+			-O "./target/fungible_token-0.1.3.wasm";\
+	fi
+	@echo ⚙️ Running all tests...
+	@cargo +nightly t -- --include-ignored --test-threads=1
```
Comment (fluiderson) [inline Makefile:59]: Please make test sections & deps downloading like here: https://github.com/gear-dapps/game-of-chance/blob/013736d757e56f4a180b3d7e160818a4c2c976ed/Makefile#L29-L56.

File: state/src/lib.rs [hunk 6]
```diff
@@ -0,0 +1,8 @@
+use io::StakingMetadata;
+use gmeta::{metawasm, Metadata};
+use gstd::{prelude::*, ActorId};
+
+#[metawasm]
+pub trait Metawasm {
+    type State = <StakingMetadata as Metadata>::State;
+}
```
Comment (fluiderson) [inline state/src/lib.rs:8]: Why are there no functions?

File: tests/node_test.rs [hunk 7]
```diff
@@ -0,0 +1,130 @@
+use gclient::{EventProcessor, GearApi, Result};
+use gstd::Encode;
+use staking::io::{InitStaking, StakingAction};
+
+#[cfg(debug_assertions)]
+const PATH: &str = "./target/wasm32-unknown-unknown/debug/staking.opt.wasm";
+
+#[cfg(not(debug_assertions))]
+const PATH: &str = "./target/wasm32-unknown-unknown/release/staking.opt.wasm";
```
Comment (fluiderson) [inline tests/node_test.rs:9]:
```suggestion
use staking::WASM_BINARY_OPT;
```

File: tests/node_test.rs [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+        distribution_time: 10000,
+        reward_total: 1000,
+    };
+
+    let staking_payload = staking.encode();
+
+    let gas_info = api
+        .calculate_upload_gas(
+            None,
+            gclient::code_from_os(PATH)?,
```
Comment (fluiderson) [inline tests/node_test.rs:35]:
```suggestion
            WASM_BINARY_OPT.into(),
```

File: tests/node_test.rs [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+            staking_payload.clone(),
+            0,
+            true,
+            None,
+        )
+        .await?;
+
+    let (message_id, _program_id, _hash) = api
+        .upload_program_bytes_by_path(
+            PATH,
```
Comment (fluiderson) [inline tests/node_test.rs:45]:
```suggestion
        .upload_program_bytes(
            WASM_BINARY_OPT,
```
And similarly modify the rest of this file...

**Comment by fluiderson** [comment]
And please add the `[workspace]` section with the metawasm crate like here: https://github.com/gear-dapps/app/blob/c762aeed4ce9e563bc58af5ca9115285deb4a60b/Cargo.toml#L32-L35.

**Review by LouiseMedova — APPROVED** [review]

**Pull request merged**